src/plugins/image_embeddable/public/components/image_editor/open_image_editor.tsx

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+
+import { PresentationContainer, tracksOverlays } from '@kbn/presentation-containers';
+import { toMountPoint } from '@kbn/react-kibana-mount';
+import { FilesContext } from '@kbn/shared-ux-file-context';
+
+import { ImageConfig } from '../../image_embeddable/types';
+import { FileImageMetadata, imageEmbeddableFileKind } from '../../imports';
+import { coreServices, filesService, securityService } from '../../services/kibana_services';
+import { createValidateUrl } from '../../utils/validate_url';
+import { ImageViewerContext } from '../image_viewer/image_viewer_context';
+
+export const openImageEditor = async ({
+  parentApi,
+  initialImageConfig,
+}: {
+  parentApi: PresentationContainer;
+  initialImageConfig?: ImageConfig;
+}): Promise<ImageConfig> => {
+  const { ImageEditorFlyout } = await import('./image_editor_flyout');
+
+  const { overlays, theme, i18n, http } = coreServices;
+  const user = securityService ? await securityService.authc.getCurrentUser() : undefined;
+  const filesClient = filesService.filesClientFactory.asUnscoped<FileImageMetadata>();
+
+  /**
+   * If available, the parent API will keep track of which flyout is open and close it
+   * if the app changes, disable certain actions when the flyout is open, etc.
+   */
+  const overlayTracker = tracksOverlays(parentApi) ? parentApi : undefined;
```

Comment:
By using the parent API to track the overlays, the dashboard can now (1) disable its top menu options when the image flyout is open and (2) it will also be responsible for closing the currently-open-flyout on navigation, like so:

https://github.com/elastic/kibana/assets/8698078/4d084a1d-ae01-41bd-bef1-526e415be69a